File: seer-ts/src/utils/clipboard.ts
export async function copyToClipboard(text: string): Promise<void> {
  if (!text) return;

  const platform = process.platform;

  if (platform === "darwin") {
    await runClipboard(text, "pbcopy");
    return;
  }

  if (platform === "win32") {
    await runClipboard(text, "cmd", "/c", "clip");
    return;
  }

  // Linux: try wl-copy, xclip, xsel
  const candidates = [
    ["wl-copy"],
    ["xclip", "-selection", "clipboard"],
    ["xsel", "--clipboard", "--input"],
  ];

  let lastErr: Error | null = null;
  for (const [cmd, ...args] of candidates) {
    try {
      const which = Bun.spawnSync(["which", cmd]);
      if (which.exitCode !== 0) continue;
      await runClipboard(text, cmd, ...args);
      return;
    } catch (e) {
      lastErr = e as Error;
    }
  }

  throw lastErr ?? new Error("no clipboard utility found");
}

async function runClipboard(text: string, cmd: string, ...args: string[]): Promise<void> {
  const proc = Bun.spawn([cmd, ...args], {
    stdin: "pipe",
  });
  proc.stdin.write(text);
  proc.stdin.end();
  await proc.exited;
}

File: archive/go-v0/clipboard.go
package main

import (
	"errors"
	"os/exec"
	"runtime"
	"strings"
)

func copyToClipboard(text string) error {
	if text == "" {
		return nil
	}

	switch runtime.GOOS {
	case "darwin":
		return runClipboardCommand(text, "pbcopy")
	case "windows":
		return runClipboardCommand(text, "cmd", "/c", "clip")
	default:
		candidates := [][]string{
			{"wl-copy"},
			{"xclip", "-selection", "clipboard"},
			{"xsel", "--clipboard", "--input"},
		}
		var lastErr error
		for _, c := range candidates {
			if _, err := exec.LookPath(c[0]); err != nil {
				continue
			}
			if err := runClipboardCommand(text, c[0], c[1:]...); err == nil {
				return nil
			} else {
				lastErr = err
			}
		}
		if lastErr != nil {
			return lastErr
		}
		return errors.New("no clipboard utility found (tried wl-copy, xclip, xsel)")
	}
}

func runClipboardCommand(text, name string, args ...string) error {
	cmd := exec.Command(name, args...)
	cmd.Stdin = strings.NewReader(text)
	return cmd.Run()
}
